frigate/object_detection.py

```diff
@@ -22,46 +29,150 @@ def ReadLabelFile(file_path):
     return ret
 
 class PreppedQueueProcessor(threading.Thread):
-    def __init__(self, cameras, prepped_frame_queue):
+    def __init__(self, cameras, prepped_frame_queue, tf_args):
 
         threading.Thread.__init__(self)
         self.cameras = cameras
         self.prepped_frame_queue = prepped_frame_queue
         
+        # print("tf_args: " + str(tf_args))
         # Load the edgetpu engine and labels
-        self.engine = DetectionEngine(PATH_TO_CKPT)
-        self.labels = ReadLabelFile(PATH_TO_LABELS)
+        if tf_args != None:
```

Comment:
Passing in different label files and models is already supported. You would do that with a Docker volume by passing `-v path/to/model/on/host.tflite:/frozen_inference_graph.pb` to the docker command or by specifying in your compose file for both the model and the labels.

Reply:
Yes, that is true. Passing a custom graph via docker parameter works fine. 

Let me elaborate why I suggest adding config parameters as an option:

1. It allows to add supplementary parameters that might be useful. At the moment I only added /data dir for saved samples, but I anticipate potentially inference hyperparameters as we play with different models.

2. In the next iteration I was hoping we can allow multiple models to be used in one frigate run. I saw folks asking on the forums how they can run multiple models. Different models may be applicable for different regions or cameras. My tests with Coral show that switching between models has a minimal impact of about 10ms. It seems reasonable in the bigger picture. Apparently if multiple Corals are plugged-in, the TF Lite framework knows how to load balance, cache and optimize model reuse. I haven't tried that yet myself.

Hope that makes sense. Happy to jump on a call to brainstorm a bit. Ping me when time allows: ivelin117@gmail.clom

Reply:
When the time comes to support multiple models, then I would want to add the ability to pass them as parameters. It would likely be implemented in a different way for that use case anyway. For now, I don't think it is necessary.